### PR TITLE
Escape / for script tags

### PR DESCRIPTION
**Pull request by adborden**
http://www.w3.org/TR/html-markup/script.html

Allowed contents:
http://www.w3.org/TR/html-markup/syntax.html#non-replaceable-character-data

I'm not sure if we also want to be escaping comment tags "<!--" Anyone else familiar with script escaping best practices?

### CODE REVIEW

**Comment by adborden** [comment]
Thanks @rothfels for pointing this out.

File: src/teacup.coffee [hunk 1]
```diff
@@ -174,6 +176,13 @@ class Teacup
     @raw contents
     @raw "</#{tagName}>"
 
+  scriptTag: (tagName, args...) ->
+    {attrs, contents} = @normalizeArgs args
+    @raw "<#{tagName}#{@renderAttrs attrs}>"
+    @raw @scriptescape contents
```
Comment (gwwar) [inline src/teacup.coffee]: Wouldn't you html escape this by default, and instead let the user explicitly pass a raw block into the script tag if it's safe content? It's tricky embedding a script tag with potentially dirty data. See: https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet for a quick reference. If you have an explicit example I can try breaking the template.

Comment (gwwar) [inline src/teacup.coffee]: I suppose this is ok if we remember to stringify any dirty content into the script body.

Comment (gwwar) [inline src/teacup.coffee]: Oh found it: https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#RULE_.233.1_-_HTML_escape_JSON_values_in_an_HTML_context_and_read_the_data_with_JSON.parse

Comment (gwwar) [inline src/teacup.coffee]: from: https://blog.whitehatsec.com/handling-untrusted-json-safely/
```
<span style=”display:none” id=”init_data”>
<%= data.to_json %>  <– data is HTML escaped –>
</span>
```
```
<script>
// unescapes the content of the span
var jsonText = document.getElementById(‘init_data’).innerHTML;
// parse untrusted JSON safely
var initData = JSON.parse(jsonText);
</script>
```

```
<script>
// unescapes the content of the span
var jsonText = document.getElementById(‘init_data’).innerHTML;
// parse untrusted JSON safely
var initData = JSON.parse(jsonText);
</script>
```

**Comment by adborden** [comment]
Awesome info @gwwar. So it sounds like there are changes to be made in our apps. Is this PR still useful to escape the `</script>` ? That way even if folks aren't handling data right, they won't get bit by the current behavior, which is their script content escaping the script tag.

**Comment by gwwar** [comment]
Escaping the forward slash does help, but in the long run I would suggest html escaping by default. Users should still be able to override that by using an explicit raw call.

**Comment by adborden** [comment]
Changed to HTML escaping.

cc @gwwar @bobzoller

**Comment by gwwar** [comment]
:+1:

**Pull request merged**